streaming/java/streaming-runtime/src/main/java/io/ray/streaming/runtime/worker/tasks/StreamTask.java

```diff
@@ -60,45 +64,67 @@ private void prepareTask() {
         WorkerInternalConfig.OP_NAME_INTERNAL, executionVertex.getExecutionJobVertexName());
 
     // producer
+
     List<ExecutionEdge> outputEdges = executionVertex.getOutputEdges();
-    Map<String, BaseActorHandle> outputActors = new HashMap<>();
+
+    // merge all output edges to create writer
+    List<String> outputChannelIds = new ArrayList<>();
+    List<BaseActorHandle> targetActors = new ArrayList<>();
 
     for (ExecutionEdge edge : outputEdges) {
-      String queueName = ChannelID.genIdStr(
+      String queueName = ChannelId.genIdStr(
           taskId,
           edge.getTargetExecutionVertex().getExecutionVertexId(),
           executionVertex.getBuildTime());
-      outputActors.put(queueName, edge.getTargetExecutionVertex().getWorkerActor());
+      outputChannelIds.add(queueName);
+      targetActors.add(edge.getTargetExecutionVertex().getWorkerActor());
     }
 
-    if (!outputActors.isEmpty()) {
-      List<String> channelIDs = new ArrayList<>();
-      outputActors.forEach((vertexId, actorId) -> {
-        channelIDs.add(vertexId);
+    if (!targetActors.isEmpty()) {
+      DataWriter writer = new DataWriter(
+          outputChannelIds, targetActors, jobWorker.getWorkerConfig()
+      );
+
+      // create a collector for each output operator
+      Set<String> opNameSet = new HashSet<>();
+      Map<String, List<String>> opGroupedChannelId = new HashMap<>();
+      Map<String, List<BaseActorHandle>> opGroupedActor = new HashMap<>();
+      Map<String, Partition> opPartitionMap = new HashMap<>();
+      for (int i = 0; i < outputEdges.size(); ++i) {
+        ExecutionEdge edge = outputEdges.get(i);
+        String opName = edge.getTargetExecutionVertex().getExecutionJobVertexName();
```

Comment:
Maybe a method defined in ```ExecutionEdge```: getTargetOperatorName()

Reply:
done.